File: exercise/Week1/Ex2/api-ts-node/userRepository.ts
import { UserDTO } from "./userDto";

export class UserRepository {
    private users: UserDTO[] = [
        {id: "1", email: "alex@example.com"},
        {id: "2", email: "john@example.com"}
    ];
    
    getUserByEmail(email: string) {
        return this.users.find(user => user.email === email);
    }
}

File: exercise/Week1/Ex2/.NET_mini_api/UserRepository.cs
using System.Collections.Generic;
using System.Linq;
using MiniAPI;

public class UserRepository
{
    private static readonly List<UserDto> _users = new()
    {
        new UserDto(Guid.NewGuid(), "alex@example.com"),
        new UserDto(Guid.NewGuid(), "john@example.com")
    };

    public UserDto? GetUserByEmail(string email)
    {
        return _users.FirstOrDefault(user => user.Email == email);
    }
}